core/types/cx_receipt.go

```diff
@@ -82,6 +86,36 @@ func (cs CXReceipts) MaxToShardID() uint32 {
 	return maxShardID
 }
 
+// ComputeMerkleRoot computes the merkle root of this list of receipts
+// The root is the hash of [shardID, receiptsRoot, shardID, receiptsRoot...]
+// The receiptRoot is the merkle root hash of the receipts for a specific shards.
+func (cs CXReceipts) ComputeMerkleRoot() common.Hash {
+	byteBuffer := bytes.NewBuffer([]byte{})
+	for i := 0; i <= int(cs.MaxToShardID()); i++ {
+		shardReceipts := cs.GetToShardReceipts(uint32(i))
+		if len(shardReceipts) != 0 {
+			sKey := make([]byte, 4)
+			binary.BigEndian.PutUint32(sKey, uint32(i))
+			byteBuffer.Write(sKey)
+			hash := DeriveSha(shardReceipts)
+			byteBuffer.Write(hash[:])
+		}
+	}
+	return crypto.Keccak256Hash(byteBuffer.Bytes())
```

Comment:
will it cause issue if byteBuffer.Bytes() is empty?

Reply:
no